components/eamxx/src/share/field/field_utils_impl_colred.hpp

```diff
@@ -0,0 +1,123 @@
+#ifndef SCREAM_FIELD_UTILS_IMPL_COLRED_HPP
+#define SCREAM_FIELD_UTILS_IMPL_COLRED_HPP
+
+#include "ekat/kokkos/ekat_kokkos_utils.hpp"
+#include "ekat/mpi/ekat_comm.hpp"
+#include "share/field/field.hpp"
+
+namespace scream {
+namespace impl {
+
+// Utility to compute the reduction of a field along its column dimension.
+// This is equivalent to einsum('i,i...k->...k', f1, f2); i is the column.
+// The layouts are such that:
+// - The first dimension is for the columns (COL)
+// - There can be only up to 3 dimensions
+
+template <typename ST>
+Field column_reduction(const Field &f1, const Field &f2, const ekat::Comm *co) {
```

Comment:
Some comments on the fcn interface:

- I would use a more self-documenting name than f1/f2. Maybe you could replace f1->weight (and f2 with just f)?
- I don't think you need a comm in this routine, so you can just prune it.
- Can we offer 2 versions? One like this one, and one that takes an already allocated output field. The former would create the field an call the latter, but the latter can be more efficiently used at runtime, to avoid allocs.

Reply:
But you have to, before we merge it, right? There's not much sense to the integral otherwise...